backend/pubsub.js

```diff
@@ -25,11 +25,27 @@ import type { JSONType } from '~/shared/types.js'
 const { bold } = require('chalk')
 const WebSocket = require('ws')
 
-// ====== API ====== //
-
 const { PING, PONG, PUB, SUB, UNSUB } = NOTIFICATION_TYPE
 const { ERROR, SUCCESS } = RESPONSE_TYPE
 
+// Used to tag console output.
+const tag = '[pubsub]'
+
+// ====== Helpers ====== //
+
+const generateSocketID = (() => {
+  let counter = 0
+
+  return (debugID) => String(counter++) + (debugID ? '-' + debugID : '')
+})()
+
+const log = console.log.bind(console, tag)
+log.bold = (...args) => console.log(bold(tag, ...args))
+log.debug = console.debug.bind(console, tag)
+log.error = (...args) => console.error(bold.red(tag, ...args))
```

Comment:
Do these lines end up modifying `console.log` for the rest of the server?

If so, does it make more sense to create a special function just for this file?

Reply:
> Do these lines end up modifying console.log for the rest of the server?
No they don't, calling `.bind()` creates a new function and does not mutate anything.

By the way I've been considering defining this logger in its own shared module, but it turns out that console styling works quite differently in the browser and in Node.